### PR TITLE
fix: cfd-582 - remove duplicate stops when they exist within one fare zone

### PR DESCRIPTION
**Pull request by JwanKhalaf**
# Description

-   Remove duplicate stops when they exist within one fare zone

# Testing instructions

-   Test against real data that has duplicate stops within the same fare zone

# Type of change

-   [ ] feat - A new feature
-   [x] fix - A bug fix
-   [ ] docs - Documentation only changes
-   [ ] codestyle - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] refactor - A code change that neither fixes a bug nor adds a feature
-   [ ] perf - A code change that improves performance
-   [ ] test - Adding missing tests or correcting existing tests
-   [ ] build - Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
-   [ ] ci - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] chore - Other changes that don't modify src or test files
-   [ ] revert - Reverts a previous commit
-   [ ] content - Changes to content or copy

### CODE REVIEW

File: repos/fdbt-netex-output/src/netex-convertor/point-to-point-tickets/pointToPointTicketNetexHelpers.ts [hunk 1]
```diff
@@ -33,17 +33,55 @@ export const hasDuplicates = (array: string[]): boolean => {
...
 
     const stopsAtcoCodes = stops.map(s => s.atcoCode);
 
     const hasDuplicateStops = hasDuplicates(stopsAtcoCodes);
 
-    if (hasDuplicateStops) {
+    return hasDuplicateStops;
+};
+
+export const getPointToPointScheduledStopPointsList = (fareZones: FareZone[]): ScheduledStopPoints[] => {
```
Comment (joeds13) [inline repos/fdbt-netex-output/src/netex-convertor/point-to-point-tickets/pointToPointTicketNetexHelpers.ts:62]: If there a case in the data where there would be stop duplicates within a single farezone and across farezones?

In that case this code would not remove the cross farezone duplicates

File: repos/fdbt-netex-output/src/netex-convertor/point-to-point-tickets/pointToPointTicketNetexHelpers.ts [hunk 2]
```diff
@@ -33,17 +33,55 @@ export const hasDuplicates = (array: string[]): boolean => {
     return new Set(array).size !== array.length;
 };
 
-export const getPointToPointScheduledStopPointsList = (fareZones: FareZone[]): ScheduledStopPoints[] => {
-    // we want a copy (by value, not by reference) because we do not want to alter the fare zones array being passed in
-    const copyOfFareZones = JSON.parse(JSON.stringify(fareZones));
+export const hasDuplicatesWithinASingleFareZone = (fareZones: FareZone[]): boolean => {
```
Comment (joeds13) [inline repos/fdbt-netex-output/src/netex-convertor/point-to-point-tickets/pointToPointTicketNetexHelpers.ts:36]: Do unittests exists for this?

File: repos/fdbt-netex-output/src/netex-convertor/point-to-point-tickets/pointToPointTicketNetexHelpers.ts [hunk 3]
```diff
@@ -33,17 +33,55 @@ export const hasDuplicates = (array: string[]): boolean => {
...
+        const hasDuplicateStops = hasDuplicates(atcoCodesOfStopsWithinFareZone);
+
+        return hasDuplicateStops;
+    });
+
+    return hasDuplicatesWithinASingleFareZone;
+};
 
-    let stops = getStops(copyOfFareZones);
+export const hasDuplicatesAcrossFareZones = (fareZones: FareZone[]): boolean => {
```
Comment (joeds13) [inline repos/fdbt-netex-output/src/netex-convertor/point-to-point-tickets/pointToPointTicketNetexHelpers.ts:52]: Do unittests exists for this?

**Review by joeds13 — DISMISSED** [review]

File: repos/fdbt-netex-output/src/netex-convertor/point-to-point-tickets/pointToPointTicketNetexHelpers.ts (hunk 2 above)
Comment (JwanKhalaf) [reply repos/fdbt-netex-output/src/netex-convertor/point-to-point-tickets/pointToPointTicketNetexHelpers.ts:36]: Added :+1:

File: repos/fdbt-netex-output/src/netex-convertor/point-to-point-tickets/pointToPointTicketNetexHelpers.ts (hunk 1 above)
Comment (JwanKhalaf) [reply repos/fdbt-netex-output/src/netex-convertor/point-to-point-tickets/pointToPointTicketNetexHelpers.ts:62]: I'm not sure what you mean, can you elaborate?

When there are duplicate stops within a single fare zone, we wish to remove the duplicates. In the NeTEx they would be removed from the fareZones tag and the ScheduledStopPoints tag.

I hope that makes sense.

Comment (JwanKhalaf) [reply repos/fdbt-netex-output/src/netex-convertor/point-to-point-tickets/pointToPointTicketNetexHelpers.ts:62]: Added new unit tests. :+1:

**Pull request merged**